### DISCUSSION TITLE
Get SelectedIndex from TabStrip

### DISCUSSION THREAD

**Discussion by SirZeeno** [AvaloniaUI/Avalonia#15888] [Q&A] [answered]
i have a tab strip and i want to get the selected index from it using the MVVM of writing your code, so I have a tab strip and assign the SelectedIndex variable to a binding of a get set variable in the ViewModel of the View, but nothing is being updated in the get set variable and the only way I've been able to access the SelectedIndex number is through giving the tab strip a name, for example HeaderTabs and then use HeaderTabs.SelectedIndex to get the index. Visual demonstration below.

````
public int SelectedTab
    {
        get => _Tabindex;
        set
        {
            ConsoleExt.WriteLineWithPretext($"SelectedTab changed to: {_Tabindex}", ConsoleExt.OutputType.Info);
            this.RaiseAndSetIfChanged(ref _Tabindex, value);
        }
    }
````
even tried this to see if it would change anything
````
public int SelectedTab
    {
        get => _Tabindex;
        set
        {
            if (_Tabindex == value) return;
            _Tabindex = value;
            ConsoleExt.WriteLineWithPretext($"SelectedTab changed to: {_Tabindex}", ConsoleExt.OutputType.Info);
            OnPropertyChanged();
        }
    }
````
And here is the OnPropertyChanged function I am using
````
public event PropertyChangedEventHandler PropertyChanged;

    private void OnPropertyChanged([CallerMemberName] string propertyName = null)
    {
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
    }
````

````
<TabStrip Name="HeaderTabs" SelectedIndex="{Binding SelectedTab, Mode=TwoWay}" HorizontalAlignment="Right">
                    <TabItem Header="Home" FontWeight="Heavy" Height="20" FontSize="15" IsVisible="False"></TabItem>
                    <TabItem Header="Genres" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Types" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Updated" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Added" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Ongoing" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Upcoming" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                    <TabItem Header="Watch List" FontWeight="Heavy" Height="20" FontSize="15"></TabItem>
                </TabStrip>
````
and I am using a textblock to show me the current index in the ui
````
<TextBlock Text="{Binding SelectedTab}" />
````
And lastly, this is the only thing that provides an index to me that also changes as a different tab gets selected,
````
int TabIndex = HeaderTabs.SelectedIndex
````

Any help in this is appreciated because I have no clue what could be causing this issue.

**Comment by jp2masa** [comment, marked as answer]
Did you check if the `DataContext` is correctly set, e.g. using DevTools (F12)?

**Reply by SirZeeno** [reply]
So surprisingly I set the Datacontext for the tabstrip to what I already set it for the entire window and it works now, but I've been using
````
<UserControl.DataContext>
        <vm:MainViewModel />
    </UserControl.DataContext>
````
to see the context for the entire window, but for some reason, the datacontext in devtools was defaulting to MainView instead of MainViewModel, even though I never set the datacontext or datatype as anything but MainViewModel, how could that be?

edit: I figured out the issue, it was test statement that I've written that set the datacontext to something else after the program launched, so I didn't see it until later.
